src/cli-util.js

```diff
@@ -392,9 +332,165 @@ function formatFiles(argv, filepatterns) {
+    header += ` or -${option.alias}`;
+  }
+
+  switch (option.type) {
+    case "boolean":
+      // do nothing
+      break;
+    case "choice":
+      header += ` <${option.choices
+        .filter(choice => !choice.deprecated)
+        .map(choice => choice.value)
+        .join("|")}>`;
+      break;
+    default:
+      header += ` <${option.type}>`;
+      break;
+  }
+
+  if (header.length >= threshold) {
+    header += "\n" + " ".repeat(threshold);
+  } else {
+    header += " ".repeat(threshold - header.length);
+  }
+
+  return (
+    header + option.description.replace(/\n/g, "\n" + " ".repeat(threshold))
+  );
+}
+
+function capitalize(str) {
+  return str.replace(/^[a-z]/, char => char.toUpperCase());
+}
+
+function indent(str, spaces) {
+  return str.replace(/^/gm, " ".repeat(spaces));
+}
+
+function normalizeArgv(rawArgv, options) {
+  options = options || {};
+
+  const consoleWarn = options.warning === false ? () => {} : console.warn;
+
+  const normalized = {};
+
+  Object.keys(rawArgv).forEach(key => {
+    const rawValue = rawArgv[key];
+
+    if (key === "_") {
+      normalized[key] = rawValue;
+      return;
+    }
+
+    if (key.length === 1) {
+      // do nothing with alias
+      return;
+    }
+
+    const option = constant.detailOptions[key];
+
+    if (option === undefined) {
+      // unknown option
+      return;
+    }
+
+    const value = getValue(rawValue, option);
+
+    if (option.exception !== undefined) {
+      if (typeof option.exception === "function") {
+        if (option.exception(value)) {
+          normalized[key] = value;
+          return;
+        }
+      } else {
+        if (value === option.exception) {
+          normalized[key] = value;
+          return;
+        }
+      }
+    }
+
+    switch (option.type) {
+      case "int":
+        validator.validateIntOption(value, option);
+        normalized[key] = Number(value);
+        break;
+      case "choice":
+        validator.validateChoiceOption(value, option);
+      // eslint-disable-next-line no-fallthrough
+      default:
+        normalized[key] = value;
+        break;
+    }
+  });
+
+  return normalized;
+
+  function getValue(rawValue, option) {
+    if (rawValue && option.deprecated) {
+      let warning = `\`--${option.name}\` is deprecated.`;
+      if (typeof option.deprecated === "string") {
+        warning += ` ${option.deprecated}`;
+      }
+      consoleWarn(warning);
+    }
+
+    const value = option.getter(rawValue, rawArgv);
+
+    if (option.type === "choice") {
+      const choice = option.choices.find(choice => choice.value === rawValue);
+      if (choice !== undefined && choice.deprecated) {
+        const warningDescription =
+          rawValue === ""
+            ? "without an argument"
+            : `with value \`${rawValue}\``;
+        consoleWarn(
+          `\`--${option.name}\` ${warningDescription} is deprecated. Automatically redirect to \`--${option.name}=${choice.redirect}\`.`
```

Comment:
`Automatically redirect` sounds a bit odd to me, maybe `Defaulting to` or `Falling back to`?